src/pygeoprocessing/geoprocessing.py

```diff
@@ -2506,7 +2506,11 @@ def convolve_2d(
             values in these areas will be nonsensical numbers, perhaps
             numerical infinity or NaNs.
         normalize_kernel (boolean): If true, the result is divided by the
-            sum of the kernel.
+            sum of the kernel. If ``ignore_nodata_and_edges`` is also true,
+            this divides by the sum of the area of the kernel that is valid 
+            (not nodata or edge). For instance, a 3x3 kernel centered on a 
+            corner pixel would sum the 4 pixels that are within the raster 
+            and divide by 4.This combination is useful for averaging a raster.
```

Comment:
That behavior description is correct, but it's not related to whether  `normalize_kernel` is True or False. `normalize_kernel` is only about "does your kernel sum to 1.0?" Otherwise what you describe is the functionality of `ignore_nodata_and_edges` which makes me think it should go there, but I feel like the first sentence says that it does this already:

``` 
ignore_nodata_and_edges (boolean): If true, any pixels that are equal
            to ``signal_path_band``'s nodata value or signal pixels where the
            kernel extends beyond the edge of the raster are not included when
            averaging the convolution filter.
```

But I like this idea of "how should I use these confusing parameters in this confusing function?" Would you be willing to expand this PR into making a more helpful descriptive paragraph above the args? The function does so much more than this:

```
Convolves the raster in ``kernel_path_band`` over ``signal_path_band``.
    Nodata values are treated as 0.0 during the convolution and masked to
    nodata for the output result where ``signal_path`` has nodata.
```

I think describing when you'd want to use any combination of normalize_kernel, ignore_nodata_and_edges, and mask_nodata would be nice to get up front rather than having to read the arguments.

Reply:
Sure, I guess I put it with `normalize_kernel` because that's the feature that seemed relevant for my use case of averaging the raster. Agree it would be better in the descriptive paragraph above the args. 
> Otherwise what you describe is the functionality of ignore_nodata_and_edges which makes me think it should go there, but I feel like the first sentence says that it does this already

When I read the docstring originally it wasn't clear to me if `ignore_nodata_and_edges` would apply to the normalizing behavior as well as the convolution. I looked at the code and saw that `normalize_kernel` [divides everything by `kernel_sum`](https://github.com/natcap/pygeoprocessing/blob/d8c7cab583d08d26f0d9230e7d0a12f944779526/src/pygeoprocessing/geoprocessing.py#L3372) and that [`kernel_sum` is a constant](https://github.com/natcap/pygeoprocessing/blob/d8c7cab583d08d26f0d9230e7d0a12f944779526/src/pygeoprocessing/geoprocessing.py#L3323) so I concluded it wouldn't do what I wanted. James pointed out that it actually does work but the way that happens is pretty cryptic.

> I think describing when you'd want to use any combination of normalize_kernel, ignore_nodata_and_edges, and mask_nodata would be nice to get up front rather than having to read the arguments.

That would be super helpful to have in the docstring! I don't have a good understanding of what you'd use the other combinations of options for, though.

Reply:
Hi @emlys, just getting back to this, I'm sorry if you were going to do this directly, but it sounded like you might have wanted a suggestion so I modified your PR to include that narrative about what `ignore_nodata_and_edges` does, also included another sentence in the `ignore_nodata_and_edges` parameter description and reverted the description of `normalize_kernel`. If you're okay with this, please feel free to merge the PR directly (I've already "approved") or I'm very happy to do more review if you want to work on this more.